src/QsCompiler/Transformations/CallGraph.cs

```diff
@@ -0,0 +1,569 @@
+                        if (filter.Contains(child))
+                        {
+                            if (child == startNode)
+                            {
+                                foundCycle = true;
+                                cycles.Add(nodeStack.Reverse().ToList());
+                            }
+                            else if (!blockedSet.Contains(child))
+                            {
+                                foundCycle |= populateCycles(child);
+                            }
+                        }
+                    }
+
+                    nodeStack.Pop();
+
+                    if (foundCycle)
+                    {
+                        unblock(currNode);
+                    }
+                    else
+                    {
+                        // Mark currNode as being blocked on each of its children
+                        // If any of currNode's children unblock, currNode will unblock
+                        foreach (var child in intputSCC[currNode])
+                        {
+                            if (filter.Contains(child))
+                            {
+                                if (!blockedMap.ContainsKey(child))
+                                {
+                                    blockedMap[child] = new HashSet<int>() { currNode };
+                                }
+                                else
+                                {
+                                    blockedMap[child].Add(currNode);
+                                }
+                            }
+                        }
+                    }
+
+                    return foundCycle;
+                }
+
+                populateCycles(startNode);
+
+                return cycles;
+            }
+        }
+
+        /// <summary>
+        /// This is a dictionary mapping source nodes to information about target nodes. This information is represented
+        /// by a dictionary mapping target node to the edges pointing from the source node to the target node.
+        /// </summary>
+        private readonly Dictionary<CallGraphNode, Dictionary<CallGraphNode, ImmutableArray<CallGraphEdge>>> _Dependencies =
+            new Dictionary<CallGraphNode, Dictionary<CallGraphNode, ImmutableArray<CallGraphEdge>>>();
+
+        private QsNullable<ImmutableArray<ResolvedType>> RemovePositionFromTypeArgs(QsNullable<ImmutableArray<ResolvedType>> tArgs) =>
+            tArgs.IsValue
+            ? QsNullable<ImmutableArray<ResolvedType>>.NewValue(tArgs.Item.Select(StripPositionInfo.Apply).ToImmutableArray())
+            : tArgs;
+
+        private CallGraphEdge RemovePositionFromEdge(CallGraphEdge edge) =>
+            new CallGraphEdge() { ParamResolutions = edge.ParamResolutions.ToImmutableDictionary(kvp => kvp.Key,
+                kvp => StripPositionInfo.Apply(kvp.Value)) };
+
+        private void RecordDependency(CallGraphNode callerKey, CallGraphNode calledKey, CallGraphEdge edge)
```

Comment:
This method relies on that the edge and all entries in _Dependencies do not contain any positional information. You should add a doc comment to that effect.

Reply:
This comment has not been addressed either.

Reply:
Removing position information has been moved to the `AddDependency` methods. This need to have position information stripped is documented on the CallGraphEdge class.